[FE] 프로필 페이지 - 프로필정보

ChanYangYu opened the pull request:
## 관련 이슈

- closes #65

## 셀프 체크리스트

> 최소 1명 이상의 assign을 받아야만 Merge 가능합니다.

- [ ] Warning Message 발생 유/무
- [x] Coding Convention 준수 유/무

## 작업 요약
> 작업 내역 요약
- 프로필 페이지 userSlice 데이터 적용
- 열정,나눔 스택 userSlice 데이터 적용 및 디자인 변경
- 프로필페이지 전체 디자인 변경

## Preview

> 선택 사항

![image](https://user-images.githubusercontent.com/55623688/141677518-fab8c35c-f73c-45e2-865d-79e02922158e.png)

## PR 특이 사항

> PR을 볼 때 주의깊게 봐야하거나 말하고 싶은 점
> 백로그에 없지만 필요하여 구현한 기능 

- 원래는 프로필 페이지 편집을 먼저 구현하려고했는데 커밋을 하다보니 프로필 페이지 디자인이 더 많아서 프로필 정보 issue를 먼저 끝내게 됐습니다.
- 전체적으로 디자인 다시 바꿔봤는데 쉽지않네요... 디자인 부분에서 과감하게 피드백 날려주시면 바로 적용하겠습니다!

ChanYangYu committed:
[FE] feat: 자기소개 텍스트 편집 버튼 구현


- ProfileBoxContainer : ProfileBox를 기술스택, 활동내역에서 재사용을 위해 기본 레이아웃만 담은 컴포넌트
- ProfileIntroBox : ProfileBox자기소개 내용을 편집 및 표시하는 컴포넌트
- issue (#65)

ChanYangYu committed:
[FE] fix: ProfileBoxContainer Props 변경


- ProfileBoxContainer 여러 Children을 받을 수 있도록 Props 변경
- issue (#65)

ChanYangYu committed:
[FE] feat: ProfileTechStackBox 및 임시 모달 구현 


- ProfileTechStackBox:  현재 TechStackList 출력 및 편집 Modal 관리하는 컴포넌트
- EditTechStack : Modal 내부 컴포넌트
- 편집 버튼 클릭시 모달 활성화 구현

ChanYangYu committed:
[FE] feat: ProfileActivityListBox 컴포넌트 생성


- ProfileActivityListBox: 활동내역  컴포넌트 
- ProfileBox index.ts파일 생성

ChanYangYu committed:
[FE] style: ProfileUserContents 변경된 컴포넌트 적용


- ProfileUserContents에 생성한 ProfileBox 주입
- ProfileBoxContainer 스타일 수정

ChanYangYu committed:
[FE] feat: 사용자 로그인 정보 적용 


- 기존 하드코딩 깃허브 정보 userSlice 데이터 적용
- issue (#65)

ChanYangYu committed:
[FE] feat: mini TechStackInput 모듈화 


- TechStackInput에서 바로 ReduxSlice로 상태를 변경하는 로직 대신 외부에서 state와 setState를 주입받아 재사용할 수 있도록 구현
- issue (#65)

ChanYangYu committed:
[FE] feat: CustomButton 모듈화 


- Custom Button style을 주입할 수 있도록 변경
  - color
  - backgroundColor
  - margin
- EditTechStack 컴포넌트 내부에 Custom버튼 구현

ChanYangYu committed:
[FE] feat: 멘토링 스택 정보 표시


- 멘토링 스택 정보를 표시하는 ProfileMentorStackBox 구현
- issue (#65)

ChanYangYu committed:
[FE] design:  프로필 페이지 UI 구조 변경


- SideContents 디자인 수정
- ProfileContentsBox 세부 디자인 수정
- issue (#65)

ChanYangYu committed:
[FE] refactor: 의미없는 모듈화 제거


- PrrofileUserContents, ProfileMentorContents 같이 depth만 깊게 만드는 컴포넌트 제거 및 내부 컴포넌트 통합
- issue (#65)

ChanYangYu committed:
[FE] feat: 나눔 ,열정 스택  데이터 적용


- userSlice 의 fever, share stack 유저 데이터 적용
- issue (#65)

ChanYangYu committed:
[FE] style: prettier 적용


- prettier 적용
- issue (#65)

ChanYangYu committed:
[FE] refactor: BoxModal, ProfilePage 컴포넌트로 이동


- ProfileTechStackBox에서 생성했던 BoxModal을 ProfilePage 컴포넌트에서 생성하여 전역으로 관리하도록 변경(추후 다른 ProfileBox에서 모달의 element만 바꿔서 재사용이 가능하도록 하기위함)
- issue (#65)

Donggggg commented:
#### 디자인 피드백
디자인 피드백 드려요!
1. 프로필의 각 박스의 제목의 font-weight을 높이면 좋을 것 같아요.
2. 스택 아이템 박스의 수직 padding을 줄이면 좋을 것 같아요.

ChanYangYu committed:
[FE] design: 프로필 페이지 스타일 피드백 적용


- 프로필의 박스 title font-weight bold 적용
- 기술 스택 아이템 박스의 수직 marin제거

Donggggg commented on client/src/pages/GroupCreatePage/index.tsx:
```diff
@@ -24,6 +24,7 @@ function GroupCreatePage(): JSX.Element {
   const dispatch = useAppDispatch();
 
+  const setUsingTechStacks = (newTechStacks: string[]) => dispatch(setGroupData({ usingTechStacks: newTechStacks }));
   const getContents = (): JSX.Element | null => {
```
예전에는 괜찮을 것 같았는데 다시 보니 `getContent` 함수의 case문의 숫자가 매직넘버로 판단되네요!
상수화하면 좋을 것 같습니다.

ChanYangYu replied:
> 주말까지 고생하셨습니다! 부분적으로 리뷰할 수 있는건 진행했는데 전체적으로 궁금한 점이 하나 생겼는데 `ProfilePage/modal/EditTechStack`의 `currentUsingTechStacks`을 props로 관리하는 이유가 따로 있나요? 전역 스토어에서 관리하면 더 좋을 것 같다고 생각이 들어서요!

아 이부분은 아직 데이터가 없어서 더미 테스트로 하려다보니 이러한 코드가 나온것 같네요! 이후에  프로필페이지 편집 부분에서 API연동하면서 전역으로 상태로 바꿀게요! 섬세한 리뷰 정말 감사합니다!😀

Donggggg commented on client/src/pages/ProfilePage/index.tsx:
```diff
@@ -1,17 +1,40 @@
+        <ProfileMentorStackBox />
+      </MainContents>
+
+      {isModal && (
+        <BoxModal
+          style={{
+            width: '700px',
+            height: '400px',
+            padding: '50px 50px 50px 50px',
+          }}
+          element={
+            <EditTechStack currentUsingTechStacks={[]} onCancel={handleModalBackgroundClick} />
+          }
+          onCancel={handleModalBackgroundClick}
+        />
+      )}
```
style 속성이 길어질 때는 변수로 분리하는 것도 좋아보여요!

ChanYangYu replied:
바로 적용했습니다!

Donggggg commented on client/src/pages/ProfilePage/ProfileSideContents.tsx:
```diff
@@ -2,22 +2,23 @@ import React from 'react';
         <GithubURLContainer>
-          <GithubIcon src={GithubLogo} alt="GithubIcon"></GithubIcon>
-          <GithubURL href="https://github.com/ChanYangYu">GitHub Storage</GithubURL>
+          <GithubIcon src={GithubLogo} alt="GithubIcon" />
+          <GithubURL href={`https://github.com/${user.oAuthId}`}>GitHub Storage</GithubURL>
```
styled-compoents 네이밍 직관적이어서 너무 좋은 것 같네요!

Donggggg commented on client/src/pages/ProfilePage/FeverSharingBar.tsx:
```diff
@@ -1,47 +1,59 @@
-  to {
-    width: 100px;
-  }
+const ProgressAnimation = (width: string | number | undefined) => keyframes`
+  from { width: 0; }
+  to { width: ${width}; }
```
프로그레스바 애니메이션 구현이 이렇게 깔끔하게 나오셨군요 ..

Donggggg reviewed (commented):
주말까지 고생하셨습니다!
부분적으로 리뷰할 수 있는건 진행했는데 전체적으로 궁금한 점이 하나 생겼는데
`ProfilePage/modal/EditTechStack`의 `currentUsingTechStacks`을 props로 관리하는 이유가 따로 있나요?
전역 스토어에서 관리하면 더 좋을 것 같다고 생각이 들어서요!

yuchanleeme commented on client/src/pages/ProfilePage/FeverSharingBar.tsx:
```diff
@@ -1,47 +1,59 @@
 
+const FeverNumber = styled.p`
+  font-weight: bold;
+  color: #ee7262;
```
개인적인 의견이지만 열정지수, 나눔지수 같이 자주 사용되는 색깔은 따로 theme에 지정해서 사용해도 좋을것 같습니다!!

ChanYangYu replied:
네 이부분은 확정되면 추가해서 적용하겠습니다!

yuchanleeme commented on client/src/pages/GroupCreatePage/index.tsx:
```diff
@@ -108,15 +115,21 @@ function GroupCreatePage(): JSX.Element {
         <CustomButton
+          style={{
+            color: '#00C5AA',
+            backgroundColor: '#FFFFFF',
```
CustomButton에 계속 중복되는 
``` css
color: '#00C5AA',
backgroundColor: '#FFFFFF',
```
부분은 CustomButton 컴포넌트 button style에 적용해도 괜찮지 않을까요??

yuchanleeme replied:
아니면 각 버튼의 역할이나 label 이름을 customButton 컴포넌트에게 넘겨주고, 거기에 맞는 style 값을 customButton 컴포넌트 내부에 저장한다음에 매칭 시켜서 사용해도 좋을것 같습니다!!

ChanYangYu replied:
이부분  CustomButton에 적용했습니다! 리뷰 감사합니다~

ChanYangYu committed:
[FE] style: modal style 상수화


- 바꾸지 않을 style 상수로 선언하여 사용
- issue (#65)

ChanYangYu committed:
[FE] feat: CustomButton 추상화


- CustomButton Props의 label에 따라 스타일이 결정되도록 추상화

ChanYangYu committed:
[FE] fix: CustomButton 변경사항 적용 


- CustomButton 호출 부분 수정
- GroupCreatePage 매직넘버 제거
- 그룹생성 정보에 ownerId 추가

ChanYangYu committed:
[FE] fix: 그룹생성페이지 추가 변경사항 적용


- 기술스택 선택화면 필터링 버그 수정
- 카테고리 커서 포인터
- CreateForm 박스 그림자 크기 조절,  게이지 텍스트bold
- GroupInfo 박스 그림자 조절

ChanYangYu committed:
[FE] feat: 멘토신청 버튼 구현


- 멘토가 아닌경우 멘토링스택이 아닌 멘토신청 로직이 나오도록 구현
- issue (#65)

Donggggg reviewed (approved):
LGTM
